File: easy/3921_score_validator/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  std::vector<int> scoreValidator(std::vector<std::string> &events) {
    int score = 0;
    int counter = 0;

    for (const std::string &event : events) {
      if (counter == 10)
        break;

      if (event == "W")
        ++counter;
      else if (event == "WD" || event == "NB")
        ++score;
      else
        score += event[0] - '0';
    }

    return {score, counter};
  }
};

File: easy/3921_score_validator/solution.py
from typing import List


class Solution:
    def scoreValidator(self, events: List[str]) -> List[int]:
        score = 0
        counter = 0

        for event in events:
            if event == "W":
                counter += 1
            elif event == "WD" or event == "NB":
                score += 1
            else:
                score += ord(event) - 48

            if counter == 10:
                break

        return [score, counter]
